src/channel/Channel.h

```diff
@@ -83,6 +83,13 @@ class ChannelBuilder
     }
     NodeId GetPeerNodeId() const { return mPeerNodeId; }
 
+    ChannelBuilder & SetPeerFabricId(FabricId peerFabricId)
```

Comment:
It does not make sense to me that we would have a `mPeerFabricId`, and particularly not a public setter.  Isn't the source of truth for this the fabric ID in `mOperationalCredentialSet[mOperationalCredentialSetIndex]`?  Why duplicate this information?  That seems like a good way to get into an inconsistent state.

Reply:
Updated.